src/Chunk.cpp

```diff
@@ -2076,36 +2074,73 @@ bool cChunk::DoWithEntityByID(UInt32 a_EntityID, cLambdaEntityCallback a_Callbac
 
 
 
-bool cChunk::ForEachBlockEntity(cBlockEntityCallback & a_Callback)
+// Helper function - there's probably a better place for this
+
+template <BLOCKTYPE, BLOCKTYPE...> bool IsOneOfImpl(BLOCKTYPE, std::false_type);
+template <class = void> bool IsOneOfImpl(BLOCKTYPE, std::true_type);
```

Comment:
`IsOneOf<Types ...>` always calls `IsOneOfImpl<Types ...>` which for the empty parameter pack is `IsOneOfImpl<>`. 

If it were simply `bool IsOneOfImpl(BLOCKTYPE, std::true_type)` then it wouldn't be considered in overload resolution because it isn't a template.